zapetw/core_windows.go

```diff
@@ -0,0 +1,39 @@
+package zapetw
+
+import (
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func getETWencoder() zapcore.Encoder {
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	return zapcore.NewJSONEncoder(encoderConfig)
+}
+
+func getETWCore(eventName string, loggingLevel zapcore.Level) (zapcore.Core, error) {
+	etwSyncer, err := NewETWWriteSyncer(eventName, loggingLevel)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to initialize ETW logger")
+	}
+	etwcore := zapcore.NewCore(getETWencoder(), zapcore.AddSync(etwSyncer), loggingLevel)
+	return etwcore, nil
+}
+
+func InitETWLogger(eventName string, loggingLevel zapcore.Level) (*zap.Logger, error) {
+	etwcore, err := getETWCore(eventName, loggingLevel)
+	if err != nil {
+		return nil, err
```

Comment:
can we wrap error and return errors.wrap() here as well like getETWCore()

Reply:
The errors are already wrapped within the internal methods to provide specific context at their source. Initially I had chosen not to wrap the errors again.